substrate/client/network/src/litep2p/mod.rs

```diff
@@ -163,6 +163,9 @@ pub struct Litep2pNetworkBackend {
 	/// Pending `PUT_VALUE` queries.
 	pending_put_values: HashMap<QueryId, (RecordKey, Instant)>,
 
+	/// Pending `GET_PROVIDERS` queries.
+	pending_get_providers: HashMap<QueryId, (RecordKey, Instant)>,
```

Comment:
nit: We could merge `pending_get_values`, `pending_put_values` and `pending_get_providers` into a single `pending_queries`.

The new `pending_queries` can hold sufficient information to distinguish between query types:

```rust
pending_queries: HashMap<QueryId, QueryType>

enum QueryType {
    GetValue((RecordKey, Instant)),
    PutValue((RecordKey, Instant)),
    GetProviders((RecordKey, Instant)),
}
```

The advantage of doing so is that we would remove duplicated code that handles the case where the queryID is not found in our mappings. At the same time, we will never forget about checking a hashmap basically turning code like:

```
if let Some(..) = pending_get_values.get(..) {

} else if let Some(..) = pending_put_values.get(..) {

} else if let Some(..) = pending_queries.get(..) {
    ..
}

```
Into a more easier to follow:

```
match pending_queries.get(..) {
    None => // absent,
    Some(GetValue(..))
    ...
}
```

Up to you in the end